### ISSUE TITLE
Unable to import in Next 15 due to registerApolloClient

### ISSUE THREAD

**Issue by oalexdoda** [apollographql/apollo-client-nextjs#383]
Hi,

It looks like `@apollo/experimental-nextjs-app-support` is throwing an error where it can't find `registerApolloClient`. 

Not sure if it's because of it importing from `index.ssr.js` vs. another file, or what I'm doing wrong.

Would appreciate some help!

Thanks!

Code:

![image](https://github.com/user-attachments/assets/e75ef099-d76e-4688-a219-290363681ecc)

Server Error:

![image](https://github.com/user-attachments/assets/5feb537d-a134-4131-90ec-9608a65d30da)

**Comment by oalexdoda** [comment]
I had to pass the use server directive in that file for the import to work and it no longer crashes.

![image](https://github.com/user-attachments/assets/7eb5bdcc-ea17-42bc-9e27-0654c024473e)

but it does cause a type error here

![image](https://github.com/user-attachments/assets/d29d1c1a-af3d-425c-8779-cc377373247e)

it builds/works in dev, but not sure if that's how it's supposed to be.

here's the index.ts that exports things (this is all in a transpiled next.js package)

![image](https://github.com/user-attachments/assets/dcd9ebeb-ffd2-4ade-96f3-290614675b28)

and how it gets exported in the package.json

![image](https://github.com/user-attachments/assets/27fe5364-2af5-4144-aad4-dafebd1a131f)

not sure if any of this helps, but trying to give you some context of how it's used

**Comment by phryneas** [comment]
Yeah, you definitely shouldn't use the "use server", it turns everything there into server functions, not server components.

I assume that your `index.ts` here is the problem - if you import *anything* from a barrel file like that into a `"use client"` file, that turns *everything* there into "client side" code, which will then crash, even if it was only meant for the server and never for the client.

I'm afraid, but with RSC/Next.js you can't use barrel files like that - you either have to create a `index.rsc.js` and `index.cc.js` (or similar), or - probably even better for your bundle size - stop using barrel files altogether.

**Comment by oalexdoda** [comment]
Got it, that makes perfect sense - thanks a lot @phryneas !
